apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClientImpl.cs

```diff
@@ -49,6 +49,12 @@ public sealed partial class SubscriberClientImpl : SubscriberClient
     public SubscriberClientImpl(SubscriptionName subscriptionName, IEnumerable<SubscriberServiceApiClient> clients, Settings settings, Func<Task> shutdown)
         : this(subscriptionName, clients, settings, shutdown, TaskHelper.Default) { }
 
+    // _modifyDeadlineSeconds is the AckDeadline in seconds before which the message must be acknowledged or have its lease extended. If the user hasn't specified a value of AckDeadline, the default of 60 seconds is used.
```

Comment:
Rather than specifying the details of each field, I was expecting to see comments right at the top (even before the class declaration) about the *conventions* for field names:

- An "ackDeadline" is the amount of time we extend the lease by in each auto-extend request
- A suffix of "window" is the amount of time we allow for an auto-extend request to take place
- A suffix of "interval" is a time we wait between sending one auto-extend request and the next; this is derived from the ackDeadline and the window.

... and then the rules about what can be done with each value:

- An ack deadline can be limited by min/max ack deadlines
- A window can be limited by min/max windows
- An interval can be calculated as "ack deadline - window"
- An interval can be limited by min/max intervals

Any other computation is forbidden.

(I'm considering experimenting with enshrining this in the type system, with some internal structs...)

Reply:
Thanks. This rule part makes it clear what you and Amanda are thinking.